DataModel/SubSample.h

```diff
@@ -3,27 +3,48 @@
 
 #include <iostream>
 #include <vector>
+#include <stdint.h>
 
 class SubSample{
 
  public:
 
-  SubSample() {};
+  /// Type for relative hit times within a SubSample. Unit = ns
+  typedef float relative_time_t;
 
-  SubSample(std::vector<int> PMTid,std::vector<float> time);
+  /// Type for absolute timestamps of a SubSample. Unit = us
+  typedef int64_t timestamp_t;
 
-  SubSample(std::vector<int> PMTid, std::vector<float> time, std::vector<float> charge);
+  /// Factor between unit of timestamp and hit times
+  static const relative_time_t s_TIMESTAMP_TO_RELATIVE_TIME = 1e3; //  = 1 us/ns
 
-  std::vector<int> m_PMTid;
-  std::vector<float> m_time;
-  std::vector<float> m_charge;
-  std::vector<int> m_time_int;
-  std::vector<int> m_charge_int;
+  SubSample() {};
+  SubSample(std::vector<int> PMTid, std::vector<relative_time_t> time);
```

Comment:
Should be able to set the timestamp in this version too (currently, I believe that no tools are actually using the charge information, although it is filled by WCSimReader)
```suggestion
  SubSample(std::vector<int> PMTid, std::vector<relative_time_t> time, timestamp_t timestamp = 0);
```

Reply:
Same as above. Was trying to avoid default values and use overloading to keep everything consistent.